persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxGenerator.java

```diff
@@ -176,48 +178,35 @@ public static void populateEntities(Module.Builder moduleBuilder, SyntaxTree bal
                     .typeDescriptor();
             for (Node node : recordDesc.fields()) {
                 EntityField.Builder fieldBuilder;
-                String qualifiedNamePrefix = null;
-                if (node.kind() == SyntaxKind.RECORD_FIELD_WITH_DEFAULT_VALUE) {
-                    RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) node;
-
-                    fieldBuilder = EntityField.newBuilder(fieldNode.fieldName().text().trim());
-                    String fType;
-                    TypeDescriptorNode type;
-                    if (fieldNode.typeName().kind().equals(SyntaxKind.ARRAY_TYPE_DESC)) {
-                        type = ((ArrayTypeDescriptorNode) fieldNode.typeName()).memberTypeDesc();
+                RecordFieldNode fieldNode = (RecordFieldNode) node;
```

Comment:
Cant this logic be improved similar to https://github.com/ballerina-platform/module-ballerina-persist/blob/main/compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistModelDefinitionValidator.java#L235-L295

As far as I can see, this logic too is trying to access the type and process if it is optional or array type.

Reply:
updated in [c7dda48](https://github.com/ballerina-platform/persist-tools/pull/145/commits/c7dda48fdf8cf74c92d55ac0b96df0c4b9d30471)